client/components/Weather.jsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState, useEffect, useRef } from 'react';
+import axios from 'axios';
+import { Typography, Box } from '@mui/material';
+
+const Weather = () => {
+  const [weatherData, setWeatherData] = useState(null);
+  const weatherRef = useRef(weatherData);
+
+  // Get api/weather
+  const getWeatherData = () => {
+    axios.get('api/weather')
+    .then((response) => {
+      // set state using setWeatherData
+      setWeatherData(response.data);
+    })
+    .catch((err) => {
+      console.error('Error getting weather data', err);
+    })
+  };
+
+
+  useEffect(() => {
+    getWeatherData();
+  }, [weatherRef]);
+
+  // conversion to Fahrenheit
+  const toFahrenheit = (celsius) => {
+    return (celsius * 9/5) + 32;
```

Comment:
Use Math.floor(((celsius * 9/5) + 32) * 100) / 100 to round to the nearest hundredth of a degree